### ISSUE TITLE
User POV listing

### ISSUE THREAD

**Issue by quancore** [markus-wa/demoinfocs-golang#114]
**Higher goal**
At any time, which players are in a POV (point of view) of another player stored as a list of player entities.

**Describe the solution you'd like**
This feature is already implemented on another parser namely spotters and spotted. Check: https://saul.github.io/demofile/classes/_entities_player_.player.html#hasspotted

**Comment by markus-wa** [comment]
Good idea.

It will probably be a while before I get to this. In the meanwhile you could try to use `Player.Entity` and read the `m_bSpottedByMask` property (see saul's parser on how to use it)

**Comment by quancore** [comment]
Could be an event triggering when a player has entered the POV of player, like onEnterPov, or when close onExitPOV. At least there would be a struct of a list like (POV player, tick to enter POV).

**Comment by markus-wa** [comment]
Yeah such an event might make sense. (This should also be possible right now with `Player.Entity.FindProperty("m_bSpottedByMask").OnUpdate(...)`).

**Comment by quancore** [comment]
Is it possible for this addition on near time? Because it is a bit urgent and it would be wonderful to add. Thank you.

**Comment by markus-wa** [comment]
Unlikely, I'm too busy with work. As I said you can do everything manually already with FindProperty.

What would you consider near time? A week? month?

**Comment by quancore** [comment]
It would be wonderful to add within a week however, if your workflow is high, no luck. Thank you for your help.

**Comment by markus-wa** [comment]
Ok, I'll see if I can find time.

**Comment by markus-wa** [comment]
I implemented it in the branch `player-spotted`, could you give it a try?

You can use `Player.IsSpottedBy(Player) bool`, `Player.HasSpotted(Player) bool`, `Participants.SpottersOf(Player) []*Player` & `Participants.SpottedBy(Player) []*Player`.

It might not be ready to be merged for a while though (breaking changes and some stuff isn't documented or cleaned up).

**Comment by quancore** [comment]
I will give a try but is it also add the additional information to spotting like in which tick player spotted by or spot someone? Maybe adding a spotting struct and storing the additional info in it can be useful. In this way, spotting object can be returned and at any time it can be extended.

**Comment by markus-wa** [comment]
> More importantly, it would be wonderful to an event like onPlayerSpotted that includes spotter and spotted player (if it is easily appliable).

Yes, you're right, I forgot about that.

> which tick player spotted by or spot someone?

This should be done in your code once the events are available 🙂

**Comment by quancore** [comment]
> > More importantly, it would be wonderful to an event like onPlayerSpotted that includes spotter and spotted player (if it is easily appliable).
> 
> Yes, you're right, I forgot about that.
> 
> > which tick player spotted by or spot someone?
> 
> This should be done in your code once the events are available

Yes, you are right. Just need the event if okay. Thank you.

**Comment by markus-wa** [comment]
Something like this should work now:
```go
parser.RegisterEventHandler(func(e events.PlayerSpottersChanged) {
    fmt.Println(e.Spotted.Name, "is now spotted by:", parser.GameState().Participants().SpottersOf(e.Spotted))
})
```

**Comment by quancore** [comment]
I am getting error "panic: runtime error: invalid memory address or nil pointer dereference" at the line: github.com/markus-wa/demoinfocs-golang/common/player.go:123 using IsSpottedBy for the demo: https://www.hltv.org/matches/2296883/cloud9-vs-fnatic-esl-esea-pro-league-season-1-finals (overpass)

**Comment by markus-wa** [comment]
Fixed in 45880d085ab49a7b50e96d5cfd626bc1834d9c4a
